### PR TITLE
add `OpenMP` support

### PR DESCRIPTION
**Pull request by psychocoderHPC**
add `find_package(OpenMP)` to `CMakeLists.txt`

### CODE REVIEW

File: src/picongpu/CMakeLists.txt [hunk 1]
```diff
@@ -268,6 +268,18 @@ if( (Boost_VERSION EQUAL 105500) AND
       "${CUDA_NVCC_FLAGS} \"-DBOOST_NOINLINE=__attribute__((noinline))\" ")
 endif()
 
+################################################################################
+# OPENMP
```
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: Spelling: OpenMP

File: src/picongpu/CMakeLists.txt [hunk 2]
```diff
@@ -268,6 +268,18 @@ if( (Boost_VERSION EQUAL 105500) AND
...
+##############################################################################
+
+FIND_PACKAGE(OpenMP)
+IF(NOT OPENMP_FOUND)
+    MESSAGE(STATUS "Disable OpenMP: Can't find library")
+
```
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: pls remove this line

File: src/picongpu/CMakeLists.txt [hunk 3]
```diff
@@ -268,6 +268,18 @@ if( (Boost_VERSION EQUAL 105500) AND
       "${CUDA_NVCC_FLAGS} \"-DBOOST_NOINLINE=__attribute__((noinline))\" ")
 endif()
 
+################################################################################
+# OPENMP
+##############################################################################
+
+FIND_PACKAGE(OpenMP)
```
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: we use all lowercase spelling

**Comment by ax3l** [comment]
@BenjaminW3 this pull contains several issues:
- typos
- wrong cmake styles

and harms our concept, that an independent person should review the pull (you provided the code, @psychocoderHPC selected you for the review). Not a big problem, but let us try to avoid that pls :)

File: src/picongpu/CMakeLists.txt [hunk 4]
```diff
@@ -268,6 +268,18 @@ if( (Boost_VERSION EQUAL 105500) AND
...
+IF(NOT OPENMP_FOUND)
+    MESSAGE(STATUS "Disable OpenMP: Can't find library")
+
+ELSE()
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
```
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: this is a copy-paste from a stackoverflow list. one should use `add_definitions()` instead.

**Comment by psychocoderHPC** [comment]
sry my fault: I assigned him :-(

File: src/picongpu/CMakeLists.txt (hunk 4 above)
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: I am wrong, it's correct

File: src/picongpu/CMakeLists.txt [hunk 5]
```diff
@@ -268,6 +268,18 @@ if( (Boost_VERSION EQUAL 105500) AND
       "${CUDA_NVCC_FLAGS} \"-DBOOST_NOINLINE=__attribute__((noinline))\" ")
 endif()
 
+################################################################################
+# OPENMP
+##############################################################################
+
+FIND_PACKAGE(OpenMP)
+IF(NOT OPENMP_FOUND)
+    MESSAGE(STATUS "Disable OpenMP: Can't find library")
```
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: CMake should issue a message on it's own on that, can you verify?

File: src/picongpu/CMakeLists.txt (hunk 4 above)
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: note for later `target`-redesign: [can be added](http://cmake.3232098.n2.nabble.com/link-errors-using-OpenMP-tp7581939p7581970.html) to `COMMPILE_FLAGS` and `LINK_FLAGS`

**Comment by ax3l** [comment]
that is no problem, I am just a cmake enthusiast :)

**Pull request merged**